zipkin-common/src/test/scala/com/twitter/zipkin/storage/DependencyStoreSpec.scala

```diff
@@ -188,6 +188,51 @@ abstract class DependencyStoreSpec extends JUnitSuite with Matchers {
     result(store.getDependencies(today + 2 * day, Some(day))) should be(empty)
   }
 
+  @Test def getDependencies_intermediateSpans() = {
+    val day = MILLISECONDS.convert(1, DAYS)
+    val today = Time.now.floor(Duration.fromMilliseconds(day)).inMillis
+    val zipkinWeb = Endpoint(172 << 24 | 17 << 16 | 3, 8080, "zipkin-web")
+    val zipkinQuery = Endpoint(172 << 24 | 17 << 16 | 2, 9411, "zipkin-query")
+    val zipkinJdbc = Endpoint(172 << 24 | 17 << 16 | 2, 0, "zipkin-jdbc")
+
+    val trace = ApplyTimestampAndDuration(List(
+      Span(20L, "get", 20L,
+        annotations = List(
+          Annotation(today * 1000, Constants.ServerRecv, Some(zipkinWeb)),
+          Annotation((today + 350) * 1000, Constants.ServerSend, Some(zipkinWeb)))),
+      Span(20L, "call", 21L, Some(20L),
+        binaryAnnotations = List(
+          BinaryAnnotation(Constants.LocalComponent, "depth2", Some(zipkinWeb)))),
```

Comment:
Well, this test is artificially constructed to allow stores to "cheat" (i.e. they pass it by coincidence). So technically it does not introduce a new feature, but if we strictly follow its _intent_ , i.e. building A->B link when there are local spans between A and B, then I can easily construct traces that will fail in most stores. And if we want Anorm to pass after those _harder_ traces, it would be a store-specific feature.